### PR TITLE
ImageCard: Selectable and disabled variants, new styling for error variant

### PR DESCRIPTION
**Pull request by ymouzakis**
**This change:** (check at least one)

- [x] Adds a new feature
- [ ] Fixes a bug
- [ ] Improves maintainability
- [ ] Improves documentation
- [ ] Is a release activity

**Is this a breaking change?** (check one)

- [ ] Yes
- [x] No

**Is the:** (complete all)

- [x] Title of this pull request clear, concise, and indicative of the issue number it addresses, if any?
- [x] Test suite(s) passing?
- [x] Code coverage maximal?
- [x] Changeset added?

**What does this change address?**
This PR adds three new features to the image card:

1. New `selectable` and `collection-selectable` image card variants. There is also a `subtle-select` prop that more discretely shows the card's selectability.
2. New image card 'disables' states
3. New styling for the error state

All three of these (and some of their combinations) are visible here:

Screen Shot 2022-07-07 at 4 52 51 PM

**How does this change work?**

- Added new selectable variants to image card
- Added props for disabled and sublte-select
- Used variants and props to render checkbox
- Added onMouseEnter/Exit listeners to handle those events
- Use events, props, and classMap to use relevant classes for each variant's events

**Additional context**
Add any other context here.

### CODE REVIEW

File: packages/pharos/src/components/image-card/pharos-image-card.wc.stories.mdx
Comment (ymouzakis) [inline packages/pharos/src/components/image-card/pharos-image-card.wc.stories.mdx:126]: I will trim this file down before merge. I wanted to temporarily create all combinations of props/variants to prove it was working.

File: packages/pharos/src/components/image-card/pharos-image-card.ts
Comment (ymouzakis) [inline packages/pharos/src/components/image-card/pharos-image-card.ts:342]: I think would be better to remove the links from the card title and thumbnail when cards can be selected instead of overriding the link....

Comment (ymouzakis) [reply packages/pharos/src/components/image-card/pharos-image-card.ts:342]: I'm building confidence that leaving the links as is and using `event.preventDefault();` to avoid the link navigation is a good way to go. I feel that way because we want the styling and red on hover styling a link provides. Still not 100% convinced though. Very open to suggestions here!

File: packages/pharos/src/components/image-card/pharos-image-card.scss
Comment (ymouzakis) [inline packages/pharos/src/components/image-card/pharos-image-card.scss:136]: The linter was complaining that this needed to move up in the file..

**Comment by ymouzakis** [comment]
I still need to write tests and make a change set.

File: packages/pharos/src/components/image-card/pharos-image-card.wc.stories.mdx
Comment (daneah) [inline packages/pharos/src/components/image-card/pharos-image-card.wc.stories.mdx:319]:
```suggestion
    <pharos-image-card title="Selectable" link="#" variant="selectable-collection">
```

File: packages/pharos/src/components/image-card/pharos-image-card.scss
Comment (daneah) [inline packages/pharos/src/components/image-card/pharos-image-card.scss:205]: This SCSS isn't quite following the existing BEM approach the project uses. Based on the way this is all being used I think something like the following is more in line, requiring similar changes in the DOM:

```scss
.card__link--image.card__link--selectable {
  background-color: var(--pharos-color-marble-gray-94);
}

.card--selectable.card--hover {
  background-color: var(--pharos-color-marble-gray-base);
  transition: background-color var(--pharos-transition-duration-short) ease-in-out;
}

.card__container--error.card__container--selected {
  background-color: var(--pharos-color-white);
  transition: background-color var(--pharos-transition-duration-short) ease-in-out;
}

.card__container--error.card__container--selectable.card__container--hover {
  background-color: inherit;
  transition: background-color var(--pharos-transition-duration-short) ease-in-out;
}
```

File: packages/pharos/src/components/image-card/pharos-image-card.scss
Comment (daneah) [inline packages/pharos/src/components/image-card/pharos-image-card.scss:39]: This is currently being used as a bit of a utility class, which doesn't follow BEM very well; consider creating e.g. a `.card__link--collection.card__link--selected` and `.card__link--image.card__link--selected` split of this.

**Review by daneah — COMMENTED** [review]
Overall this is looking really good; I have a couple of suggestions on improving the CSS with an eye toward BEM.

File: packages/pharos/src/components/image-card/pharos-image-card.scss
Comment (daneah) [reply packages/pharos/src/components/image-card/pharos-image-card.scss:39]: The BEM philosophy takes a different approach: classes should be used to describe component structure and variations. Some of these variations may have the same _outcome_, but are achieved through applying styles in the context of the specific component. One way to achieve both what I'm saying and what you're saying is to use a [mixin](https://sass-lang.com/documentation/at-rules/mixin) or a [placeholder/extension](https://thesassway.com/understanding-placeholder-selectors/) to reduce the repeated CSS.

Comment (ymouzakis) [reply packages/pharos/src/components/image-card/pharos-image-card.scss]: Thanks for the explanation. The mixin is working nice.

**Review by daneah — APPROVED** [review]
Thanks for all the work on this; it's looking pretty good to me! I'd like to get one more review from the team.

**Review by daneah — CHANGES_REQUESTED** [review]
@ymouzakis Please add a changeset—then this will be good to go.

**Review by daneah — APPROVED** [review]

**Pull request merged**
